doc/application/index.rst

```diff
@@ -689,27 +689,43 @@ the Zephyr tree, for example:
 
 
 
-The paths to any Kconfig files inside the structure needs to prefixed with
-$(SOC_DIR) to make Kconfig aware of the location of the Kconfig files related to
-the custom SOC.
+The Zephyr :file:`soc/Kconfig` will create the top-level
```

Comment:
Remember that `:file:` just prints the argument in a special font. It doesn't actually include the word "file" in the output. So the output here would be "The Zephyr soc/Kconfig will create the top-level", which is probably not what you want. Needs "file" added.

It might be easier just to link to it directly with zephyr_file as well.

```suggestion
The file :zephyr_file:`soc/Kconfig` will create the top-level
```

Reply:
(I'll trust you to make this update before this PR gets merged or send a follow-up)